### DISCUSSION TITLE
theme: question on how to target identifier highlight

### DISCUSSION THREAD

**Discussion by yofriadi** [helix-editor/helix#11504] [Q&A] [answered]
hi, I use some theme but have trouble on some highlight because it is blend in with my terminal themes

this is how it looks like
![Screenshot 2024-08-16 at 07 57 26](https://github.com/user-attachments/assets/4e86cc6a-cff3-4d4a-b8de-77b49b978012)

when use visual block
![Screenshot 2024-08-16 at 07 57 48](https://github.com/user-attachments/assets/4eb3039e-2986-4ba9-ac14-255a0998017f)

see the word `http` gets blend in with terminal theme.
when debugging with treesitter, it is called `identifier`, looking at theme documentation it is not show any matching keyword.
any suggestion or direction to change this?
thank you in advance.

**Comment by the-mikedavis** [comment, marked as answer]
You can use the `:tree-sitter-highlight-name` command to figure out the theme key under the cursor. `identifier` is the name of the node in the syntax tree. Syntax highlight works by running a tree-sitter query like https://github.com/helix-editor/helix/blob/master/runtime/queries/go/highlights.scm over the tree and the `@capturename` captures in that file are the theme keys. So an `identifier` might be mapped to a theme key of `variable`, `function` or something else depending on the context it's used in

**Reply by yofriadi** [reply]
thank you for your answer! this is a knowledge that I needed to play around with theme.
